Modules/signalmodule.c

```diff
@@ -611,7 +611,11 @@ static PyObject *
 signal_strsignal_impl(PyObject *module, int signalnum)
 /*[clinic end generated code: output=44e12e1e3b666261 input=b77914b03f856c74]*/
 {
+#ifdef __HAIKU__
+    const char * res;
+#else
     char *res;
+#endif
```

Comment:
Just makes it always `const`. It is more correct semantically, because `res` is initialized with pointers to string literals and the content is not changed.
```suggestion
    const char *res;
```